pkg/client/client.go

```diff
@@ -62,7 +62,17 @@ func NewClient(token, apiEndpoint string) (runtimeclient.Client, error) {
 }
 
 func NewClientFromRestConfig(config *rest.Config) (runtimeclient.Client, error) {
-	config.Insecure = true
+	tc, err := config.TransportConfig()
+	if err != nil {
+		return nil, err
+	}
+	// when specifying the insecure flag with a certificate file the runtime client throws the following error:
+	// 	cannot create client: specifying a root certificates file with the insecure flag is not allowed
+	//
+	// set insecure flag only if certificate file is not present
+	if !tc.HasCA() {
+		config.Insecure = true
+	}
```

Comment:
related to my other comment, I believe that we can drop this whole part and rely on what is in the kubeconfig only.
When you take a look at in other words, when you take a look at https://github.com/kubesaw/ksctl/blob/456bd900b8da5fbc6b2de3fba49c4aaec88efd92/pkg/cmd/generate/cli_configs.go#L72-L78 we don't handle CA nor insecure flag there either.
```suggestion
```

Reply:
yeah, makes sense. Since the current implementation fails on OCP with trusted CA. Thanks for the suggestion.